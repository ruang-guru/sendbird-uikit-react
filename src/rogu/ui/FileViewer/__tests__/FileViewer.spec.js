import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import { FileViewerComponent as FileViewer } from "../index";
import { msg0, msg1 } from '../data.mock';

describe('FileViewer', () => {
  it('should display image', function() {
    const {
      sender,
      type,
      url,
      name = '',
    } = msg0;
    const { profileUrl, nickname = '' } = sender;
    const component = shallow(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
        />
    );
    const img = component.find('.sendbird-fileviewer__content__img');
    expect(img.length).toEqual(1);
    expect(img.prop('src')).toEqual(msg0.url);
  });

  it('should display video', function() {
    const {
      sender,
      type,
      url,
      name = '',
    } = msg1;
    const { profileUrl, nickname = '' } = sender;
    const component = shallow(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
      />
    );
    const video = component.find('.sendbird-fileviewer__content__video source');
    expect(video.length).toEqual(1);
    expect(video.prop('src')).toEqual(msg1.url);
  });

  it('should handle unsupported msg', function() {
    const unsupportedMsg = { sender: {} };
    const profileUrl = '';
    const nickname = '';
    const {
      sender,
      type = '',
      url = '',
      name = '',
    } = unsupportedMsg;
    const component = shallow(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
      />
    );
    const fallback = component.find('.sendbird-fileviewer__content__unsupported');
    expect(fallback.length).toEqual(1);
    const headerActions = component.find('.sendbird-fileviewer__header__right__actions');
    expect(headerActions.length).toEqual(0);
  });

  it('should do a snapshot test of the FileViewer DOM', function() {
    const {
      sender,
      type,
      url,
      name = '',
    } = msg0;
    const { profileUrl, nickname = '' } = sender;
    const component = renderer.create(
      <FileViewer
        profileUrl={profileUrl}
        nickname={nickname}
        type={type}
        url={url}
        name={name}
        onClose={() => {}}
        onDelete={() => {}}
        message={msg0}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
