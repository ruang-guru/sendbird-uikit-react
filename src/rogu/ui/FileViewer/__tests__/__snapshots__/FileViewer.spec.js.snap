// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileViewer should do a snapshot test of the FileViewer DOM 1`] = `
<div
  className="rogu-fileviewer"
>
  <div
    className="rogu-fileviewer__header"
  >
    <div
      className="rogu-fileviewer__header__left"
    >
      <div
        className="rogu-fileviewer__header__left__avatar"
      >
        <div
          className=" sendbird-avatar"
          role="button"
          style={
            Object {
              "height": "32px",
              "width": "32px",
            }
          }
          tabIndex={0}
        >
          <div
            className="sendbird-avatar-img rogu-image-renderer"
          >
            <div
              className="rogu-image-renderer__image"
              style={
                Object {
                  "backgroundImage": "url(https://static.sendbird.com/sample/profiles/profile_15_512px.png)",
                  "borderRadius": null,
                }
              }
            />
            <img
              alt=""
              className="rogu-image-renderer__hidden-image-loader"
              onError={[Function]}
              onLoad={[Function]}
              src="https://static.sendbird.com/sample/profiles/profile_15_512px.png"
            />
          </div>
        </div>
      </div>
      <div
        className="rogu-fileviewer__header__left__metadata"
      >
        <span
          className="rogu-fileviewer__header__left__sender-name sendbird-label rogu-label--h-3 sendbird-label--color-onbackground-1"
        >
          namanya sangat panjang sampe sampe overflow
        </span>
        <span
          className="rogu-fileviewer__header__left__createdat sendbird-label sendbird-label--body-1 sendbird-label--color-onbackground-2"
        >
          05/03/2020 20.43
        </span>
      </div>
    </div>
    <div
      className="rogu-fileviewer__header__right"
    >
      <div
        className="rogu-fileviewer__header__right__actions"
      >
        <a
          className="rogu-fileviewer__header__right__actions__download"
          href="https://sendbird-staging.s3.amazonaws.com/26FA710D-A7F6-4483-ABEA-1F07A07A112E/upload/n/e49e0a4111d4445a8840bd51f3c9d933.png"
          onClick={[Function]}
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className=" sendbird-icon "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "height": "24px",
                "minHeight": "24px",
                "minWidth": "24px",
                "width": "24px",
              }
            }
            tabIndex="0"
          >
            <test-file-stub />
          </div>
        </a>
        <div
          className="rogu-fileviewer__header__right__actions__delete"
        >
          <div
            className=" sendbird-icon "
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            style={
              Object {
                "height": "24px",
                "minHeight": "24px",
                "minWidth": "24px",
                "width": "24px",
              }
            }
            tabIndex="0"
          >
            <test-file-stub />
          </div>
        </div>
      </div>
      <div
        className="rogu-fileviewer__header__right__actions__close"
      >
        <div
          className=" sendbird-icon "
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={
            Object {
              "height": "24px",
              "minHeight": "24px",
              "minWidth": "24px",
              "width": "24px",
            }
          }
          tabIndex="0"
        >
          <test-file-stub />
        </div>
      </div>
    </div>
  </div>
  <div
    className="rogu-fileviewer__content"
  >
    <img
      alt="Uploaded by namanya sangat panjang sampe sampe overflow"
      className="rogu-fileviewer__content__img"
      onBlur={[Function]}
      onFocus={[Function]}
      src="https://sendbird-staging.s3.amazonaws.com/26FA710D-A7F6-4483-ABEA-1F07A07A112E/upload/n/e49e0a4111d4445a8840bd51f3c9d933.png"
      tabIndex="0"
    />
    <div
      className=" rogu-clamped-message-item-body  rogu-clamped-message-item-body--incoming rogu-clamped-message-item-body--viewer-mode  "
    >
      <div
        className="rogu-clamped-message-item-body__inner"
      >
        <span
          className="rogu-text-message-item-body__message sendbird-label sendbird-label--body-1 sendbird-label--color-onbackground-1"
        >
          sent-mail (1).png
        </span>
      </div>
    </div>
  </div>
</div>
`;
